token/js/test/e2e-2022/tokenMetadata.test.ts

```diff
@@ -0,0 +1,181 @@
+import { expect } from 'chai';
+
+import type { Connection, Signer } from '@solana/web3.js';
+import {
+    sendAndConfirmTransaction,
+    PublicKey,
+    Keypair,
+    SystemProgram,
+    Transaction,
+    TransactionInstruction,
+} from '@solana/web3.js';
+import { TOKEN_METADATA_DISCRIMINATOR, unpack } from '@solana/spl-token-metadata';
+
+import type { ExtensionType } from '../../src';
+import {
+    initializeTokenMetadata,
+    createInitializeMintInstruction,
+    getMint,
+    TOKEN_2022_PROGRAM_ID,
+    getExtensionData,
+    updateTokenMetadataField,
+    emitTokenMetadata,
+} from '../../src';
+import { TEST_PROGRAM_ID, newAccountWithLamports, getConnection } from '../common';
+
+const TEST_TOKEN_DECIMALS = 2;
+
+// TODO: Remove or move to src. Temp function so we can repack to use spl-token-metadata
+const lengthBuffer = (buffer: Buffer | Uint8Array): Buffer => {
+    const length = Buffer.alloc(4);
+    length.writeUIntLE(buffer.length, 0, 4);
+    return length;
+};
+
+describe('Token Metadata', () => {
+    // TODO @mistersimon:- Change to instruction creator once merged
+    // const EXTENSIONS = [ExtensionType.MetadataPointer];
+    // const mintLen = getMintLen(EXTENSIONS);
+    const mintLen = 234;
+
+    let connection: Connection;
+    let payer: Signer;
+    let mint: Keypair;
+    const authority = Keypair.fromSecretKey(
+        new Uint8Array([
+            118, 177, 37, 231, 15, 88, 210, 92, 79, 231, 202, 22, 11, 15, 121, 54, 95, 229, 149, 119, 48, 177, 187, 198,
+            223, 51, 225, 74, 12, 54, 172, 36, 207, 107, 122, 208, 209, 168, 61, 177, 190, 137, 23, 156, 84, 32, 34, 82,
+            158, 176, 55, 51, 236, 66, 130, 167, 118, 31, 120, 107, 100, 192, 147, 10,
+        ])
+    );
+
+    before(async () => {
+        connection = await getConnection();
+        payer = await newAccountWithLamports(connection, 1000000000);
+    });
+
+    beforeEach(async () => {
+        mint = Keypair.generate();
+
+        // TODO @mistersimon:- Change to instruction creator once merged
+        const metadataPointerInstruction = new TransactionInstruction({
```

Comment:
Will switch to the functions created in https://github.com/solana-labs/solana-program-library/pull/5805, just wanted to decouple them for now and keep them smaller